src/sentry/lang/native/minidump.py

```diff
@@ -28,8 +29,10 @@ def process_minidump(minidump, cfi=None):
         return ProcessState.from_minidump_buffer(minidump.read(), cfi)
     elif isinstance(minidump, TemporaryUploadedFile):
         return ProcessState.from_minidump(minidump.temporary_file_path(), cfi)
-    else:
+    elif isinstance(minidump, six.binary_type) and minidump.startswith('MDMP'):
         return ProcessState.from_minidump_buffer(minidump, cfi)
+    else:
+        return ProcessState.from_minidump(minidump, cfi)
```

Comment:
Was this intentional? This was changed recently in https://github.com/getsentry/sentry/pull/10527 and might easily cause bad behavior.